### ISSUE TITLE
TypeVars may go unsolved when using a TypeVar with a default

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#4667]
**Describe the bug**
TypeVarLikes with a default type parameter can apparently go unsolved when it should surely go to the default if it's not supplied.

**To Reproduce**
```python
from enum import Enum

class Type(Enum):
	Individual = 1
	...  # other members

class ID[TypeT: Type = Type]:
    def __init__(
        self,
        id: int,
        *,
        type: TypeT = ...,
    ): ...

ID_ZERO = ID(0, type=Type.Individual)  # type should be ID[Literal[Type.Individual]]
ARBITRARY_ID = ID(1234567890)  # type should be ID[Type]
```
both currently are `ID[Type]`.

**Expected behavior**
I've spoken to Jelle about this and he agrees that it seems like a bug in the implementation of PEP 696. I can potentially add an example of this to the PEP if necessary. 

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/50501825/220206555-1d55e7d0-942a-4361-a761-7c03b153d045.png)

**VS Code extension or command-line**
pylance: v2023.2.30

**Comment by erictraut** [comment]
The behavior is as designed. Pyright's constraint solver never produces literals as a solution unless a literal is required to meet the constraints of the call. Many common use cases would break if it always assumed you wanted the solution to be a literal type.

**Comment by Gobot1234** [comment]
Is there a way to have this work though if I do add an overload?

**Comment by erictraut** [comment]
Oh, there is a bug here in that the "may go unsolved" warning shouldn't be emitted when a TypeVar has a default. I'll fix that.

**Comment by erictraut** [comment]
I've addressed the bug that you mentioned in the title. I'm a bit confused, however, because the body of your bug report seems to speak to a different behavior that you are reporting as a bug (the fact that the constraint solver doesn't retain literals unless needed to meet the constraints). As I mentioned, the latter behavior is "as designed".

**Comment by Gobot1234** [comment]
I thought the TypeVar default was just being ignored by the type checker hence it was reporting the error. I would just like a way to express the behaviour where specifying the type kwarg would make TypeT constrained to the same type.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.295, which I just published. It will also be included in a future release of pylance.

**Comment by Gobot1234** [comment]
Thank you.

I can see that mypy also has the same behaviour of not turning `TypeT` into a literal, where would be the downsides in doing so for enums like this?

**Comment by erictraut** [comment]
The downside is that it would break a lot of existing code and would be inconsistent with what most users would expect if they pass a literal argument to a generic call target.

**Comment by Gobot1234** [comment]
> The downside is that it would break a lot of existing code and would be inconsistent with what most users would expect if they pass a literal argument to a generic call target.

If the TypeVar is covariant would this be an issue?
